static/app/components/core/button/index.tsx

```diff
@@ -144,196 +114,132 @@ interface ButtonPropsWithAriaLabel extends BaseButtonProps {
-  | HrefLinkButtonPropsWithChildren
-  | HrefLinkButtonPropsWithAriaLabel;
-
-function BaseButton({
-  size = 'md',
-  to,
-  replace,
-  preventScrollReset,
-  busy,
-  href,
-  title,
-  icon,
-  children,
-  'aria-label': ariaLabel,
-  borderless,
-  translucentBorder,
-  priority,
-  disabled,
-  type = 'button',
-  tooltipProps,
-  onClick,
-  analyticsEventName,
-  analyticsEventKey,
-  analyticsParams,
-  ...buttonProps
-}: ButtonProps) {
+const useButtonFunctionality = (props: ButtonProps | LinkButtonProps) => {
   // Fallbacking aria-label to string children is not necessary as screen
   // readers natively understand that scenario. Leaving it here for a bunch of
   // our tests that query by aria-label.
   const accessibleLabel =
-    ariaLabel ?? (typeof children === 'string' ? children : undefined);
+    props['aria-label'] ??
+    (typeof props.children === 'string' ? props.children : undefined);
 
   const useButtonTrackingLogger = () => {
     const hasAnalyticsDebug = window.localStorage?.getItem('DEBUG_ANALYTICS') === '1';
-    const hasCustomAnalytics = analyticsEventName || analyticsEventKey || analyticsParams;
+    const hasCustomAnalytics =
+      props.analyticsEventName || props.analyticsEventKey || props.analyticsParams;
     if (!hasCustomAnalytics || !hasAnalyticsDebug) {
       return () => {};
     }
 
     return () => {
       // eslint-disable-next-line no-console
       console.log('buttonAnalyticsEvent', {
-        eventKey: analyticsEventKey,
-        eventName: analyticsEventName,
-        priority,
-        href,
-        ...analyticsParams,
+        eventKey: props.analyticsEventKey,
+        eventName: props.analyticsEventName,
+        priority: props.priority,
+        href: 'href' in props ? props.href : undefined,
+        ...props.analyticsParams,
       });
     };
   };
 
   const useButtonTracking =
     HookStore.get('react-hook:use-button-tracking')[0] ?? useButtonTrackingLogger;
+
   const buttonTracking = useButtonTracking({
-    analyticsEventName,
-    analyticsEventKey,
+    analyticsEventName: props.analyticsEventName,
+    analyticsEventKey: props.analyticsEventKey,
     analyticsParams: {
-      priority,
-      href,
-      ...analyticsParams,
+      priority: props.priority,
+      href: 'href' in props ? props.href : undefined,
+      ...props.analyticsParams,
     },
     'aria-label': accessibleLabel || '',
   });
 
   const handleClick = useCallback(
     (e: React.MouseEvent<HTMLButtonElement | HTMLAnchorElement>) => {
       // Don't allow clicks when disabled or busy
-      if (disabled || busy) {
+      if (props.disabled || props.busy) {
         e.preventDefault();
         e.stopPropagation();
         return;
       }
 
       buttonTracking();
-      onClick?.(e);
+      // @ts-expect-error at this point, we don't know if the button is a button or a link
+      props.onClick?.(e);
     },
-    [disabled, busy, onClick, buttonTracking]
+    [props, buttonTracking]
```

Comment:
`props` is a new object on every render, so doing memoization on it doesn’t achieve anything. Can we please remove the `useCallback` ?

Reply:
Removed it. I will remove this entire function btw, and inline the busy/disabled checks